tests/core/job_test.py

```diff
@@ -241,59 +249,70 @@ def test__eq__false(self):
 class JobSchedulerTestCase(TestCase):
     @setup
     def setup_job(self):
-        self.scheduler = Turtle()
-        run_collection = Turtle()
-        node_pool = Turtle()
-        self.job = job.Job(
-            "jobname",
-            self.scheduler,
-            run_collection=run_collection,
-            node_pool=node_pool,
-        )
-        self.job_scheduler = job.JobScheduler(self.job)
+        self.scheduler = scheduler.ConstantScheduler()
+        run_collection = jobrun.JobRunCollection(20)
+        mock_graph = mock.Mock(autospec=True)
 
-    def test_restore_job_state(self):
-        run_collection = mocks.MockJobRunCollection(
-            get_scheduled=lambda: ['a'],
+        mock_graph.get_action_map.return_value = {}
+        node_pool = node.NodePool.from_node(
+            node=mock.Mock(name="foo", autospec=True)
         )
-        self.job_scheduler.job = Turtle(runs=run_collection)
-        self.job_scheduler._set_callback = Turtle()
-        state_data = 'state_data_token'
-        self.job_scheduler.restore_state(state_data)
-        assert_call(self.job_scheduler.job.restore_state, 0, state_data)
-        assert_length(self.job_scheduler._set_callback.calls, 1)
-        assert_call(self.job_scheduler._set_callback, 0, 'a')
+
+        mock_graph.action_map = {}
+        self.job = mock.Mock(autospec=True)
+        self.job.allow_overlap = False
+        self.job.max_runtime = datetime.timedelta(days=1)
+        self.job_scheduler = job.JobScheduler(job=self.job)
+
+    def test_restore_state_sets_job_runs(self):
+        self.job.enabled = False
+        mock_runs = [mock.Mock(), mock.Mock()]
+        job_state_data = {'runs': mock_runs, 'enabled': True}
+
+        self.job_scheduler._set_callback = lambda x: x
+
+        self.job.runs.runs = collections.deque()
+        self.job.runs.get_scheduled.return_value = [mock.Mock()]
+        self.job.get_job_runs_from_state.return_value = mock_runs
+
+        with mock.patch(
+            'tron.core.job.recovery.launch_recovery_actionruns_for_job_runs'
+        ) as mock_launch_recovery:
+            mock_launch_recovery.return_value = mock.Mock(autospec=True)
+            self.job_scheduler.restore_state(job_state_data, mock.Mock())
+            assert self.job.runs.runs == collections.deque(mock_runs)
+
+    def test_restore_job_state_launches_recovery_runs(self):
```

Comment:
I guess you can just add an assertion about this in the test above, and get rid of this test?